### DISCUSSION TITLE
"Lane guidance" and "Speeding alerts"?

### DISCUSSION THREAD

**Discussion by greymda** [alexanderlavrushko/BLE-HUD-navigation-ESP32#15] [General]
Hi,

Sygic as the app has these 2 useful functions:
- lane guidance when it's a tricky junction ahead, and
- speeding alerts, when you're above the posted speed limit (obvously)

Are these functions supported by HUD? If not, would it be possible?
Thank you for your work!

**Comment by alexanderlavrushko** [comment]
These are good features, but currently the application doesn't send this information to Bluetooth device.

**Reply by seriojel** [reply]
oh, i remember seeing speeding warnings on demo when using Sygic with websocket instead of BLE. i thought it is being passed here as well.

**Comment by seriojel** [comment]
@alexanderlavrushko  how can i see what data is sending iPhone to ESP32? is it possible to get into serial monitor? currently i can only see data length.
thank you

**Reply by alexanderlavrushko** [reply]
@seriojel you can add in `MyCharWriteCallbacks`:
```
// after this line
Serial.print("New value. ");

// add these lines
Serial.print("length = ");
Serial.print(value.length());
Serial.print(": ");
for (int i = 0; i < value.length(); ++i)
{
    char tmp[4] = "";
    sprintf(tmp, "%02X ", value[i]);
    Serial.print(tmp);
}
Serial.println();
```
Also you can remove these lines:
```
Serial.print("Reading basic data: length = ");
Serial.println(currentData.length());
```

**Reply by seriojel** [reply]
thank you for prompt reply! i have a Garmin HUD device and wonder if i can somehow connect it to Sygic.
